tests/test_connect_url.py

```diff
@@ -0,0 +1,51 @@
+"""
+Test if the explict hostname is supplied correctly to the container
+"""
+import os
+import sys
+import pytest
+import requests
+import time
+from repo2docker.app import Repo2Docker
+
+def test_env_yml(tmpdir):
+    tmpdir.chdir()
+    #q = tmpdir.join("environment.yml")
+    #q.write("dependencies:\n"
+    #        "  -  notebook==5.6.0rc1")
+    p = tmpdir.join("requirements.txt")
+    p.write("notebook==5.6.0rc1")
+
+    app = Repo2Docker()
+    argv = [str(tmpdir), ]
+    app.initialize(argv)
+    app.run = False
+    app.start()  # This just build the image and does not run it.
+    container = app.start_container()
+    port = app.port
+    hostname = app.hostname
+    # wait a bit for the container to be ready
+    container_url = 'http://{}:{}/api'.format(hostname, port)
+    # wait a bit for the container to be ready
+    # give the container a chance to start
+    time.sleep(1)
+    try:
+        # try a few times to connect
+        success = False
+        for i in range(1, 4):
+            container.reload()
+            assert container.status == 'running'
+            try:
+                info = requests.get(container_url).json()
+            except Exception as e:
+                print("Error: %s" % e)
+                time.sleep(i * 3)
+            else:
+                print(info)
+                success = True
+                break
+        assert success, "Notebook never started in %s" % container
+    finally:
+        # stop the container
```

Comment:
In this test, we will want to make sure that our "custom_display_url" is included in the output. You can see the logs with:

```python
logs = c.logs().decode('utf8', 'replace')
```

So we want to check:

```python
expected_url = "http://{}:{}".format(app.hostname, app.port)
logs = c.logs().decode('utf8', 'replace')
assert expected_url in logs
```